development-styleguide.md

```diff
@@ -11,12 +11,55 @@ Furthermore, it shows commons mistakes or scenarios and how to handle them
 - Such practice should generally be avoided as it can lead to maintenance problems and can produce hard to find bugs
 - If you don't own it, don't modify it
 
-
 ```typescript
-let bob = { name : "Bob", surname: "Smith", age : 23 }
+let bob = { name: "Bob", surname: "Smith", age: 23 };
 bob.age = 24; // avoid
 // create a new object using spread operator to update only specific fields
-const olderBob = { ...bob, age: 24 }
+const olderBob = { ...bob, age: 24 };
+```
+
+#### Avoid hiding exceptions
+
+Usually in Javascript and Typescript is hard to figure out if a function throws an exception. Try to be explicit about the exceptions you throw.
+
+```typescript
+//
+// ✅good: exception is thrown explicitly
+//
+
+const config = (key: string): Option<string> => (
+  fromNullable(process.env[key])
+);
+
+const callService () => {
```

Comment:
even better would be to document the function, esp. if the function gets imported from a library: in Typescript jumping to the code from your IDE will get you to the type definitions file, not to the source, so you'll not be able to see the actual throws unless they're documented.

Reply:
Comments are liabilities ;)
Said that, I think we can add some principles about _practical code comments_, 

Back to the guideline, do you think we can improve this definition?

Reply:
not sure, I would say to not throw exceptions altogether, just return an `Either`

Reply:
regarding documentation as a liability, I agree on those cases where documentation doesn't add any value (i.e. it's redundant) - in all other cases, when documentation describes what is not obvious, the documentation is actually an asset and thus it needs to be maintained as part of the code base, see [this discussion](https://trello.com/c/1CzV7LPb) I've started.

Reply:
If we have some mechanism to make comments in sync with the actual behavior of the code, then we can trust it, of course. I leave this to the broader discussion you mentioned.

About exceptions: I think we have broader thoughts about this subject, I'll summarize in a comment below.